doc/credentials.md

```diff
@@ -0,0 +1,229 @@
+# Credentials
+Camera credentials are stored in the EdgeX Secret Store and referenced by MAC Address. All 
+devices by default are configured with credentials from `DefaultSecretPath` unless configured
+as part of a group within `AppCustom.CredentialsMap`.
+
+Three things must be done in order to add an authenticated camera to EdgeX:
+- Add device to EdgeX
+  - Manually or via auto-discovery
+- Add `Credentials` to `Secret Store`
+  - Manually or via utility scripts
+- Map `Credentials` to devices
+  - Manually or via utility scripts
+  - Configure as `DefaultSecretPath`
+
+## Terminology / Definitions
+- **Secret**: A generic object which is stored as key/value pairs in the `Secret Store` under a specific `Secret Path` key.
+- **Credentials**: A specific type of `Secret` which contains a mapping of `username`, `password`, and authentication `mode`.
+- **Secret Store**: The place EdgeX stores all `Secrets`
+  - In secure mode this is `Vault`
+  - In non-secure mode this is `Consul` and/or `configuration.toml` via `Writable.InsecureSecrets`
```

Comment:
The secrets just end up in consul in non-secure mode from the configuration.toml. So may be this sentence needs to be modified a bit.

Reply:
Ok. I tweaked it a bit. let me know your thoughts.